src/provider/application.js

```diff
@@ -141,9 +141,9 @@ class Application extends EventEmitter {
   */
   launch() {
     if (window.self !== window.top) {
-      // 1: Setup listeners for all incoming communication and beforeunload
+      // 1: Setup listeners for all incoming communication and unload
       window.addEventListener('message', this.handleConsumerMessage);
-      window.addEventListener('beforeunload', this.unload);
```

Comment:
Do @mhemesath or @grneggandsam have any memory of why this wasn't implemented with unload originally? 
I feel like it was discussed before I had to go around adding 'download' tags everywhere in the Patient Portal, but the idea was vetoed.

Reply:
I don't recall the reason, but I feel like there was one. Did you verify that the events were sending even if we switched this to unload?

Reply:
Actually i think it dawned on me. Before unload happens when the browser starts navigating away... unload happens right before it does.  Example, if you click a link and it takes 3 seconds to load, you won't hear this event until rightbefore the page navigates away 3 seconds later. With beforeUnload.. it will trigger an event as soon as the browser starts pulling the resources.

Reply:
I have tested in all the browsers other than IE for in-page redirects, pagelet redirects just to make sure we are not braking any existing workflows, everything seems fine.
I have also tested href links with tel, sms, fax,mailto in all the browsers, this change fixes the issue(flickering).

Reply:
Update: This fails when there is a delay in server side to respond. As Mike [mentioned](https://github.com/cerner/xfc/pull/26#discussion_r194171304) in the example it won't listen to the event until rightbefore the page navigates away. We are going with the previous approach to detect if the link  is opening a telephone number, email, fax, etc.